### PR TITLE
Print events

### PR DESCRIPTION
**Pull request by petersteneteg**
Added a print function to events to help debugging

### CODE REVIEW

File: include/inviwo/core/interaction/events/keyboardevent.h [hunk 1]
```diff
@@ -34,6 +34,8 @@
 #include <inviwo/core/interaction/events/interactionevent.h>
 #include <inviwo/core/interaction/events/keyboardkeys.h>
 #include <inviwo/core/util/constexprhash.h>
+#include <fmt/format.h>
+#include <fmt/ostream.h>
```
Comment (martinfalk) [inline include/inviwo/core/interaction/events/keyboardevent.h]: why these includes?

File: include/inviwo/core/interaction/events/resizeevent.h [hunk 2]
```diff
@@ -67,6 +67,8 @@ class IVW_CORE_API ResizeEvent : public Event {
     virtual uint64_t hash() const override;
     static constexpr uint64_t chash();
 
+	virtual void print(std::ostream& ss) const override;
```
Comment (martinfalk) [inline include/inviwo/core/interaction/events/resizeevent.h]: clang format

File: include/inviwo/core/interaction/events/wheelevent.h [hunk 3]
```diff
@@ -57,6 +57,8 @@ class IVW_CORE_API WheelEvent : public MouseInteractionEvent {
     virtual uint64_t hash() const override;
     static constexpr uint64_t chash() { return util::constexpr_hash("org.inviwo.WheelEvent"); }
 
+	virtual void print(std::ostream& ss) const override;
```
Comment (martinfalk) [inline include/inviwo/core/interaction/events/wheelevent.h]: clang format

File: src/core/interaction/events/gestureevent.cpp [hunk 4]
```diff
@@ -74,4 +75,12 @@ void GestureEvent::setScreenPosNormalized(dvec2 posNorm) { screenPosNorm_ = posN
...
 
+void GestureEvent::print(std::ostream& ss) const {
+    util::printEvent(ss, "GestureEvent", std::make_pair("state", state_),
+                     std::make_pair("type", type_), std::make_pair("pos", screenPosNorm_),
+                     std::make_pair("depth", depth_), std::make_pair("canvasSize", canvasSize_),
+                     std::make_pair("deltaPos", deltaPos_),
+                     std::make_pair("modifiers", modifiers_));
+}
+
 }  // namespace inviwo
```
Comment (martinfalk) [inline src/core/interaction/events/gestureevent.cpp]: missing newline

File: src/core/interaction/events/resizeevent.cpp [hunk 5]
```diff
@@ -62,4 +64,9 @@ void ResizeEvent::setPreviousSize(size2_t previousSize) { previousSize_ = previo
 
 uint64_t ResizeEvent::hash() const { return chash(); }
 
+void ResizeEvent::print(std::ostream& ss) const {
+    util::printEvent(ss, "ResizeEvent", std::make_pair("size", size_),
+                     std::make_pair("prev", previousSize_));
+}
+
```
Comment (martinfalk) [inline src/core/interaction/events/resizeevent.cpp:71]: missing newline

File: src/core/interaction/events/mouseevent.cpp [hunk 6]
```diff
@@ -50,4 +51,12 @@ void MouseEvent::setState(MouseState state) { state_ = state; }
...
 
+void MouseEvent::print(std::ostream& ss) const {
+    util::printEvent(ss, "MouseEvent", std::make_pair("state", state_),
+                     std::make_pair("button", button_), std::make_pair("pos", pos()),
+                     std::make_pair("depth", depth()), std::make_pair("size", canvasSize()),
+                     std::make_pair("sState", buttonState()),
+                     std::make_pair("modifiers", modifiers_));
+}
+
 }  // namespace inviwo
```
Comment (martinfalk) [inline src/core/interaction/events/mouseevent.cpp]: missing newline

**Review by martinfalk — APPROVED** [review]

File: src/core/interaction/events/gestureevent.cpp (hunk 4 above)
Comment (r-englund) [reply src/core/interaction/events/gestureevent.cpp]: All of our templates for code generation are missing newlines :) https://github.com/inviwo/inviwo/blob/master/tools/templates/

Comment (petersteneteg) [reply src/core/interaction/events/gestureevent.cpp]: those are deprecated. All the new ones in tools/meta/templates has newlines...

**Pull request merged**